File: app/src/test/java/com/leaguechampions/utils/ReflectionUtils.java
package com.leaguechampions.utils;

import java.lang.reflect.Field;

public class ReflectionUtils {

    public static Object getField(Object object, String fieldName) {
        try {
            Field field = getDeclaredField(object.getClass(), fieldName);
            field.setAccessible(true);
            return field.get(object);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public static void setField(Object object, String fieldName, Object value) {
        try {
            Field field = getDeclaredField(object.getClass(), fieldName);
            field.setAccessible(true);
            field.set(object, value);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    /**
     * Recursive call
     */
    private static Field getDeclaredField(Class clazz, String fieldName) {
        Field field = null;
        Class<?> superClass = clazz.getSuperclass();
        if (superClass != null) {
            field = getDeclaredField(clazz.getSuperclass(), fieldName);
        }
        if (field == null) {
            try {
                field = clazz.getDeclaredField(fieldName);
            } catch (Exception ignored) {}
        }
        return field;
    }
}

File: libraries/core/src/test/java/com/leaguechampions/libraries/core/utils/ReflectionUtils.kt
package com.leaguechampions.libraries.core.utils

import java.lang.reflect.Field

object ReflectionUtils {

    fun getField(clazz: Any, fieldName: String): Any {
        val field = getDeclaredField(clazz.javaClass, fieldName)
        field!!.isAccessible = true
        return field.get(clazz)
    }

    fun setField(clazz: Any, fieldName: String, value: Any) {
        val field = getDeclaredField(clazz.javaClass, fieldName)
        field!!.isAccessible = true
        field.set(clazz, value)
    }

    /**
     * Recursive call
     */
    private fun getDeclaredField(clazz: Class<*>, fieldName: String): Field? {
        var field: Field? = null
        val superClass = clazz.superclass
        if (superClass != null) {
            field = getDeclaredField(superClass, fieldName)
        }
        if (field == null) {
            try {
                field = clazz.getDeclaredField(fieldName)
            } catch (ignored: Exception) {}
        }
        return field
    }
}
